simulationdataschema/model_method.py

```diff
@@ -34,19 +34,1646 @@
+        | `'NCA'`           | https://journals.aps.org/prb/abstract/10.1103/PhysRevB.47.3553 |
+        | `'OCA'`           | https://journals.aps.org/prb/abstract/10.1103/PhysRevB.47.3553 |
+        | `'slave_bosons'`  | https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.57.1362 |
+        | `'hubbard_I'`     | https://iopscience.iop.org/article/10.1088/0953-8984/24/7/075604 |
+        """,
+    )
+
+    n_impurities = Quantity(
+        type=np.int32,
+        description="""
+        Number of impurities mapped from the correlated atoms in the unit cell. This defines whether
+        the DMFT calculation is done in a single-impurity or multi-impurity run.
+        """,
+    )
+
+    n_orbitals = Quantity(
+        type=np.int32,
+        shape=["n_impurities"],
+        description="""
+        Number of correlated orbitals per impurity.
+        """,
+    )
+
+    orbitals_ref = Quantity(
+        type=OrbitalsState,
+        shape=["n_orbitals"],
+        description="""
+        References to the `OrbitalsState` sections that contain the orbitals information which are
+        relevant for the `DMFT` calculation.
+
+        Example: hydrogenated graphene with 3 atoms in the unit cell. The full list of `AtomsState` would
+        be
+            [
+                AtomsState(chemical_symbol='C', orbitals_state=[OrbitalsState('s'), OrbitalsState('px'), OrbitalsState('py'), OrbitalsState('pz')]),
+                AtomsState(chemical_symbol='C', orbitals_state=[OrbitalsState('s'), OrbitalsState('px'), OrbitalsState('py'), OrbitalsState('pz')]),
+                AtomsState(chemical_symbol='H', orbitals_state=[OrbitalsState('s')]),
+            ]
+
+        The relevant orbitals for the TB model are the `'pz'` ones for each `'C'` atom. Then, we define:
+
+            orbitals_ref= [OrbitalState('pz'), OrbitalsState('pz')]
+
+        The relevant impurities information can be accesed from the parent AtomsState sections:
+            impurity_state = orbitals_ref[i].m_parent
+            index = orbitals_ref[i].m_parent_index
+            impurity_position = orbitals_ref[i].m_parent.m_parent.positions[index]
+        """,
+    )
+
+    # ? Improve this with `orbitals_ref.occupation` and possibly a function?
+    n_electrons = Quantity(
+        type=np.float64,
+        shape=["n_impurities"],
+        description="""
+        Initial number of valence electrons per impurity.
+        """,
+    )
+
+    inverse_temperature = Quantity(
+        type=np.float64,
+        unit="1/joule",
+        description="""
+        Inverse temperature = 1/(kB*T).
+        """,
+    )
+
+    magnetic_state = Quantity(
+        type=MEnum("paramagnetic", "ferromagnetic", "antiferromagnetic"),
+        description="""
+        Magnetic state in which the DMFT calculation is done. This quantity can be obtained from
+        `orbitals_ref` and their spin state.
+        """,
+    )
```

Comment:
Is this unique to DMFT? I think this should go with the non-collinear specifications.

Reply:
I added a comment, but yes. This should be obtained by other means (in fact, you see below something defined like `resolve_magnetic_state`, but I believe this should go into `ModelSystem` by checking what are the `AtomsState` in the atomic cell.

Reply:
I removed the function and also added something there to remember we have to pass this into ModelSystem